packages/lib/server/getLuckyUser.ts

```diff
@@ -41,47 +58,19 @@ async function leastRecentlyBookedUser<T extends Pick<User, "id" | "email">>({
     {}
   );
 
-  const bookings = await prisma.booking.findMany({
-    where: {
-      AND: [
-        {
-          eventTypeId,
-        },
-        {
-          attendees: {
-            some: {
-              email: {
-                in: availableUsers.map((user) => user.email),
-              },
-            },
-          },
-        },
-      ],
-    },
-    select: {
-      id: true,
-      createdAt: true,
-      attendees: {
-        select: {
-          email: true,
-        },
-      },
-    },
-    orderBy: {
-      createdAt: "desc",
+  const attendeeUserIdAndAtCreatedPair = allBookings.reduce(
```

Comment:
`allBookings` has the same bookings as the query before used to return. The only difference is that `allBookings` also includes bookings where `userId` is in `availableUsers`. However, this doesn't matter because it's filtered out with: `if (!booking.attendees.map((attendee) => attendee.email).includes(user.email)) return;`